bundle/CMakeLists.txt

```diff
@@ -0,0 +1,163 @@
+endif()
+
+if(NOT BUILD_SHARED_LIBS)
+  set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared libraries." FORCE)
+endif()
+
+if(NOT CMAKE_INSTALL_PREFIX)
+  set(CMAKE_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}/install" CACHE STRING
+      "Install path prefix, prepended onto install directories." FORCE)
+endif()
+
+project(gtclang-all C CXX)
+enable_testing()
+cmake_minimum_required(VERSION 3.9)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../cmake")
+
+include(GTClangCloneRepository)
+
+set(MB_GIT_URL "https:///github.com/cosunae/msbuild.git" 
+    CACHE PATH "URL of the msbuild git repository to clone")
+set(MB_GIT_BRANCH "master" CACHE STRING "Branch of the msbuild git repository to clone")
+
+gtclang_clone_repository(NAME msbuild URL ${MB_GIT_URL} BRANCH ${MB_GIT_BRANCH} SOURCE_DIR GTCLANG_MSBUILD_SOURCE_DIR )
+
+list(APPEND CMAKE_MODULE_PATH "${GTCLANG_MSBUILD_SOURCE_DIR}/cmake")
+include(msbuildCMakeInit)
+
+include(msbuildAddTargetCleanAll)
+include(msbuildCheckInSourceBuild)
+include(msbuildGetArchitectureInfo)
+include(msbuildGetCompilerInfo)
+include(msbuildGetPlatformInfo)
+include(msbuildMakeStringPair)
+include(msbuildReportResult)
+
+# Include all options (including all Dawn and GTClang options)
+#include(GTClangAllOptions)
+
+msbuild_get_compiler_info()
+msbuild_get_platform_info()
+msbuild_get_architecture_info()
+
+# Prevent in source builds
+msbuild_check_in_source_build()
+
+# Add custom targets
+msbuild_add_target_clean_all(
+  "${CMAKE_BINARY_DIR}/prefix"
+  "${CMAKE_BINARY_DIR}/thirdparty"  
+  "${CMAKE_BINARY_DIR}/dawn-cmake"  
+  "${CMAKE_BINARY_DIR}/dawn"
+  "${CMAKE_BINARY_DIR}/dawn-prefix"
+  "${CMAKE_BINARY_DIR}/gtclang"
+  "${CMAKE_BINARY_DIR}/gtclang-prefix"
+  "${CMAKE_BINARY_DIR}/msbuild-cmake"
+  "${CMAKE_BINARY_DIR}/boost"
+  "${CMAKE_BINARY_DIR}/boost-prefix"
+)
+
+# Output summary of the configuration
+macro(make_config_string FIRST SECOND)
+  msbuild_make_string_pair(${FIRST} ": ${SECOND}" 20 out)
+  list(APPEND config_info ${out})
+endmacro()
+make_config_string("Platform" ${MSBUILD_PLATFORM_STRING})
+make_config_string("Architecture" ${MSBUILD_ARCHITECTURE_STRING})
+make_config_string("Compiler" ${MSBUILD_COMPILER_STRING})
+make_config_string("Build type" ${CMAKE_BUILD_TYPE})
+msbuild_report_result("Configuration summary" ${config_info})
+
+include(GTClangAddDependency)
+include(msbuildFindPackage)
+
+include(msbuildSetDownloadDir)
+
+# Set the default download directory (define GTCLANG_ALL_DOWNLOAD_DIR)
+msbuild_set_download_dir()
+
+msbuild_find_package(
+  PACKAGE dawn
+  REQUIRED_VARS dawn_DIR
+  ADDITIONAL
+    DOWNLOAD_DIR ${MSBUILD_DOWNLOAD_DIR}
+    GIT_REPOSITORY "https://github.com/cosunae/dawn.git"
+    GIT_TAG "bundle_f" 
+    MSBUILD_ROOT "${GTCLANG_MSBUILD_SOURCE_DIR}"
+    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+)
+
+gtclang_add_dependency(NAME "Boost")

```

Comment:
This functionality should be moved to MSBuild as it is also used in dawn

Reply:
I removed it, same as with dawn, since I thought this indirection was not providing much help